### PR TITLE
Add facility to specify document number on the qualification report

### PR DESCRIPTION
**Pull request by james-smith-za**
In this PR I add the ability to specify the document number for the output report, as a CLI option when running the qualification tests. I figured that an actual release may be something that we want to do manually, or if we do it with Jenkins then it'll be easier to pass along as a CLI option than as a parameter in an ini file.

I used the following command to generate the report (scroll right to see the relevant argument):
```
spead2_net_raw pytest -v qualification --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest --report-docnum=E1200-foo-bar
```

Attached here: 
[report.pdf](https://github.com/ska-sa/katgpucbf/files/10456552/report.pdf)

**Note**: There is also some LaTeX re-formatting in this PR, latexindent was a bit too eager to help with this, and I was too lazy to make the one-line change that I needed to do in the preamble, in another editor. But the formatting changes are all in a single commit, so you can easily exclude that from the review.

Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] (N/A) If dependencies are added/removed: update `setup.cfg` and `.pre-commit-config.yaml`
- [x] (N/A) If modules are added/removed: use sphinx-apidoc to update files in `doc/`
- [x] Ensure copyright notices are present and up-to-date
- [x] If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date
- [x] (N/A) If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match

Closes NGC-874.

### CODE REVIEW

**Comment by bmerry** [comment]
A high-level question before I look at it line-by-line: is there a particular reason you want to bake in the report number at the time you run the test? I would imagine wanting to assign a number to a report at the time the report is generated, even if it's for a test we ran earlier.

**Comment by james-smith-za** [comment]
> A high-level question before I look at it line-by-line: is there a particular reason you want to bake in the report number at the time you run the test? I would imagine wanting to assign a number to a report at the time the report is generated, even if it's for a test we ran earlier.

This is a fair point. I guess I could make it a cli option for the PDF generator.

**Comment by james-smith-za** [comment]
Moved the functionality to `generate_pdf.py`, now run with the command: 
```
qualification/report/generate_pdf.py --report-docid=E1200-bar-foo report.json report.pdf
```

[report.pdf](https://github.com/ska-sa/katgpucbf/files/10457125/report.pdf)

File: qualification/report/generate_pdf.py [hunk 1]
```diff
@@ -875,11 +876,17 @@ def main():
     parser.add_argument("input", help="Output of pytest --report-log=...")
     parser.add_argument("pdf", help="PDF file to write")
     parser.add_argument("-c", "--commit-id", action="store_true", help="Output commit ID of katgpucbf image")
+    parser.add_argument(
+        "--report-docid",
```
Comment (bmerry) [inline qualification/report/generate_pdf.py:880]:
```suggestion
        "--report-doc-id",
```
For consistency with `--commit-id` (and it being treated as two words in variable names).

**Review by bmerry — APPROVED** [review]
Looks good. Do we need something similar for revision? Or will we just churn out a new document ID each time, always at rev 1?

**Pull request merged**